python/src/trezorlib/cardano.py

```diff
@@ -108,16 +103,85 @@ def create_input(input) -> messages.CardanoTxInputType:
 
 
 def create_output(output) -> messages.CardanoTxOutputType:
-    if not output.get("amount") or not (output.get("address") or output.get("path")):
+    if not output.get("amount") or not (
+        output.get("address") or (output.get("addressType") is not None)
+    ):
         raise ValueError("The output is missing some fields")
```

Comment:
This is too hard to audit. What about creating the exception and then a series of ifs raising it?
```
incomplete_output = ValueError("The output is missing some fields")
if ...:
    raise incomplete_output
if ...:
    raise incomplete_output
```

Reply:
I've refactored this whole part a bit to make it more obvious. I didn't like the idea of creating an error object if there's a chance it won't be used.

Reply:
Much better now.